core-datastore-test/src/main/kotlin/com/hedvig/android/core/datastore/TestPreferencesDataStore.kt

```diff
@@ -0,0 +1,40 @@
+package com.hedvig.android.core.datastore
+
+import androidx.datastore.core.DataStore
+import androidx.datastore.preferences.core.PreferenceDataStoreFactory
+import androidx.datastore.preferences.core.Preferences
+import kotlinx.coroutines.CoroutineScope
+import java.io.File
+
+private const val testPreferencesDatastoreFileName = "hedvig_test_data_store_preferences"
+
+// https://cs.android.com/androidx/platform/frameworks/support/+/androidx-main:datastore/datastore-preferences/src/main/java/androidx/datastore/preferences/PreferenceDataStoreFile.kt;l=37-38
+private const val preferencesDatastoreFileExtension = ".preferences_pb"
+
+// https://cs.android.com/androidx/platform/frameworks/support/+/androidx-main:datastore/datastore/src/main/java/androidx/datastore/DataStoreFile.kt;l=35-36
+private const val preferencesDataStoreFileDirectory = "datastore"
+
+/**
+ * @param datastoreTestFileDirectory you probably want to use [org.junit.rules.TemporaryFolder] to generate this file.
+ * Sample:
+ * ```
+ * @get:Rule
+ * val testFolder = TemporaryFolder()
+ *
+ * // Later in the test setup
+ * TestPreferencesDataStore(
+ *   datastoreTestFileDirectory = testFolder.newFolder("datastoreTempFolder"),
+ *   coroutineScope = coroutineScope,
+ * )
+ * ```
+ */
+class TestPreferencesDataStore(
+  datastoreTestFileDirectory: File,
+  coroutineScope: CoroutineScope,
+) : DataStore<Preferences> by PreferenceDataStoreFactory.create(
+  scope = coroutineScope,
+  produceFile = {
+    val fileNameWithDatastoreExtension = "$testPreferencesDatastoreFileName$preferencesDatastoreFileExtension"
+    File(datastoreTestFileDirectory, "$preferencesDataStoreFileDirectory/$fileNameWithDatastoreExtension")
+  },
+)
```

Comment:
With this we can simply use a real DataStore<Preferences> in tests, with a temporary file as described in the comment above. Super convenient and lets us avoid having to abstract some classes, but we can use the real implementation in tests with the test DataStore, super convenient that DataStore lets you do this.